### PR TITLE
Add index page for building github pages

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design

/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Add index page for building github pages. Expamle: https://nilo19.github.io/cloud-provider-azure/

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Cloud provider for Azure
```
Comment (andyzhangx) [inline docs/index.md:1]: what's the difference between `README.md` and `index.md`?

Comment (nilo19) [reply docs/index.md:1]: index.md is for `github pages` to generate a doc website

Comment (andyzhangx) [reply docs/index.md:1]: thanks, I think we may also need such index page for csi drivers.

**Pull request merged**